src/validate.ts

```diff
@@ -29,26 +28,21 @@ validator.post(
       const config = yaml.safeLoad(file)
 
       // Validate configuration.
-      joi
-        .validate(config, configSchema, {
-          abortEarly: false,
-        })
-        .then(() => {
-          res.status(200).send('Valid configuration!')
-        })
-        .catch(err => {
-          /* Compose a report */
-          const message = err.details
-            .map(
-              (detail: { path: string[]; message: string }) =>
-                `- ${detail.path.join('.')}: ${detail.message}`,
-            )
-            .join('\n')
-
-          res.status(400).send(message)
-        })
+      await configSchema.validateAsync(config, { abortEarly: false })
```

Comment:
What is the purpose of this change? Just cleaner code using `async`/`await`? Or was it somehow required for the update?

Reply:
You can use again **then/catch** blocks but for the sake of simplicity and keep the structure consistent (because everywhere **async/await** had been used) I have converted this into **async/await**. By benefiting from **isJoi** property of Joi Error object while seperating errors that will catch. 

Here Then/Catch block solution (also tests passed)
```
      configSchema.validateAsync(config, { abortEarly: false })
        .then(() => {
          res.status(200).send('Valid configuration!')
        })
        .catch(err => {
          /* Compose a report */
          const message = err.details
            .map(
              (detail: { path: string[]; message: string }) =>
                `- ${detail.path.join('.')}: ${detail.message}`,
            )
            .join('\n')

          res.status(400).send(message)
        })
```